core/app/models/concerns/spree/user_address_book.rb

```diff
@@ -76,7 +75,11 @@ def persist_order_address(order)
       # @param default set whether or not this address will show up from
       # #default_address or not
       def save_in_address_book(address_attributes, default = false)
-        return nil unless address_attributes.present?
+        if address_attributes.present?
+          address_attributes = address_attributes.with_indifferent_access
+        else
+          return nil
+        end
```

Comment:
OK. I'll fix that conditional issue. I waffled on it myself and I'm happy to have a 2nd opinion.

@jordan-brough actually convinced me to accept a hash there when we were pairing on the original implementation of this method. Since Address is readonly and equality is based upon the values of the fields and not an id, the client actually does not have to know any id. Also if they submit with an id, then the additional fields are merged into the values of the Address matching the id, which will result in a new Address being created, with a new id. It was quite counterintuitive in that case if a client passed in an Address instance, but received a different one returned from the method.

I'm happy to set up a hangout or something if you want to discuss further.